apps/calendar/js/backend/db.js

```diff
@@ -0,0 +1,470 @@
+/* jshint loopfunc: true */
+define(function(require, exports, module) {
+'use strict';
+
+var Account = require('models/account');
+var Presets = require('common/presets');
+var Local = require('provider/local');
+var Responder = require('common/responder');
+var Store = require('store/store');
+var debug = require('common/debug')('db');
+var denodeifyAll = require('common/promise').denodeifyAll;
+var nextTick = require('common/next_tick');
+var probablyParseInt = require('common/probably_parse_int');
+var uuid = require('ext/uuid');
+
+var idb = self.indexedDB;
+
+const VERSION = 15;
+
+var store = Object.freeze({
+  events: 'events',
+  accounts: 'accounts',
+  calendars: 'calendars',
+  busytimes: 'busytimes',
+  settings: 'settings',
+  alarms: 'alarms',
+  icalComponents: 'icalComponents'
+});
+
+function Db(name, app) {
+  this.app = app;
```

Comment:
after https://github.com/mozilla-b2g/gaia/pull/29783/ we will remove the need of the `app` instance